sentry/src/main/java/io/sentry/SentryOptions.java

```diff
@@ -1301,22 +1301,45 @@ public void setEnableDeduplication(final @Nullable Boolean enableDeduplication)
     Double sample(@NotNull SamplingContext samplingContext);
   }
 
+  /**
+   * Creates SentryOptions instance without initializing any of the internal parts.
+   *
+   * <p>Used by {@link NoOpHub}.
+   *
+   * @return SentryOptions
+   */
+  static @NotNull SentryOptions empty() {
+    return new SentryOptions(true);
+  }
+
   /** SentryOptions ctor It adds and set default things */
   public SentryOptions() {
-    // SentryExecutorService should be inited before any SendCachedEventFireAndForgetIntegration
-    executorService = new SentryExecutorService();
+    this(false);
+  }
+
+  /**
+   * Creates SentryOptions instance without initializing any of the internal parts.
+   *
+   * @param empty if options should be empty.
+   */
+  private SentryOptions(final boolean empty) {
+    if (!empty) {
```

Comment:
I think the heaviest objects are inited. directly during the field declaration, maybe we should move all of them here in the ctor, so we can really do a cheap ctor.

Reply:
The most heavy part is `MainEventProcessor` with `HostnameCache` that spins up a thread and `SentryExecutorService`. The rest is mainly no-op, except the serializer but i am not sure if it makes sense to complicate it any further.

Reply:
sounds good